core/templates/pages/topic-viewer-page/practice-tab/practice-tab.component.ts

```diff
@@ -102,20 +119,49 @@ export class PracticeTabComponent implements OnInit {
           this.availableSubtopics[idx].getId());
       }
     }
-    const practiceSessionsUrl = this.urlInterpolationService.interpolateUrl(
-      PracticeSessionPageConstants.PRACTICE_SESSIONS_URL, {
-        topic_url_fragment: (
-          this.urlService.getTopicUrlFragmentFromLearnerUrl()),
-        classroom_url_fragment: (
-          this.urlService.getClassroomUrlFragmentFromLearnerUrl()),
-        comma_separated_subtopic_ids: selectedSubtopicIds.join(',')
-      });
+    let practiceSessionsUrl;
+    if (this.displayArea === 'topicViewer') {
+      practiceSessionsUrl = this.urlInterpolationService.interpolateUrl(
+        PracticeSessionPageConstants.PRACTICE_SESSIONS_URL, {
+          topic_url_fragment: (
+            this.urlService.getTopicUrlFragmentFromLearnerUrl()),
+          classroom_url_fragment: (
+            this.urlService.getClassroomUrlFragmentFromLearnerUrl()),
+          comma_separated_subtopic_ids: selectedSubtopicIds.join(',')
+        });
+    } else {
+      practiceSessionsUrl = this.urlInterpolationService.interpolateUrl(
+        PracticeSessionPageConstants.PRACTICE_SESSIONS_URL, {
+          topic_url_fragment: (
+            this.topicUrl),
+          classroom_url_fragment: (
+            this.classroomUrl),
+          comma_separated_subtopic_ids: selectedSubtopicIds.join(',') });
+    }
     this.windowRef.nativeWindow.location.href = practiceSessionsUrl;
   }
 
   isAtLeastOneSubtopicSelected(): boolean {
     return this.selectedSubtopicIndices.some(item => item);
   }
+
+  getBackgroundForProgress(i: number): number {
+    return this.subtopicMasteryArray[i];
+  }
+
+  subtopicMasteryPosition(i: number): number {
+    if (this.subtopicMasteryArray[i] <= 89) {
+      return 225 - this.subtopicMasteryArray[i] * 2.5;
+    }
+    return 225 - (this.subtopicMasteryArray[i] * 2) - 15;
```

Comment:
Add a comment on how these were calculated or what these are used for.

Reply:
Done.